internal/compiler/output_columns.go

```diff
@@ -423,10 +424,12 @@ const (
 func isTableRequired(n ast.Node, col *Column, prior int) int {
 	switch n := n.(type) {
 	case *ast.RangeVar:
-		if n.Alias == nil && *n.Relname == col.Table.Name {
-			return prior
+		tableMatch := *n.Relname == col.Table.Name
+		aliasMatch := true
+		if n.Alias != nil && col.TableAlias != "" {
+			aliasMatch = *n.Alias.Aliasname == col.TableAlias
 		}
-		if n.Alias != nil && *n.Alias.Aliasname == col.TableAlias && *n.Relname == col.Table.Name {
+		if aliasMatch && tableMatch {
 			return prior
 		}
```

Comment:
If the column doesn't use a table alias, don't check that the aliases match.

Reply:
Seems correct logically, although `aliasMatch` defaulting to true is potentially confusing to future readers. I might suggest just putting this comment you wrote in the actual source.